### PR TITLE
VSCODE - Improve how the VSCode writer works and handles crashes

### PR DESCRIPTION
**Pull request by chillbrodev**
Previously there were  issues with mismatched types that were causing crashes and format exceptions. Now the code is a bit cleaner to follow and understand especially in terms of handling named configs.

Wrapped writeConfig logic in a try/catch block to handle possible comment failures and provide information for any crashes.

If there is a crash, it returns the original file content.

### CODE REVIEW

**Comment by chillbrodev** [comment]
@ibrahim-mubarak Could I get some eyes on this PR? Thanks

**Comment by ibrahim-mubarak** [comment]
I believe this will not preserve the arguments which were not part of dart defines,
such as `--profile` or `--vmservice-out-file=detective_connect.txt` 
I am unable to test this out myself as I do not have access to my main work PC right now

File: lib/src/config_writer/vscode_config_writer.dart [hunk 1]
```diff
@@ -33,34 +33,49 @@ class VscodeConfigWriter extends ConfigWriter {
   String writeConfig(String fileContent) {
     /// launch.json usually contains comments, which is valid only in JSON5.
     /// At this point however we cannot preserve these comments.
-    fileContent = fileContent.replaceAll(RegExp('.+//.+\n'), "");
```
Comment (ibrahim-mubarak) [inline lib/src/config_writer/vscode_config_writer.dart:36]: Any reason why this was removed?

File: lib/src/config_writer/vscode_config_writer.dart [hunk 2]
```diff
@@ -33,34 +33,49 @@ class VscodeConfigWriter extends ConfigWriter {
...
+    List<String> dartDefineList,
   ) {
-    return config.update(
-      'args',
-      (value) => getNonDartDefineArguments(value).followedBy(dartDefineList),
-      ifAbsent: () => dartDefineList,
-    );
+    Map<String, dynamic> configUpdate = Map<String, dynamic>.from(config);
+
+    print('ConfigUpdate: $configUpdate');
```
Comment (ibrahim-mubarak) [inline lib/src/config_writer/vscode_config_writer.dart:71]: Can you remove all the print statements?

**Review by ibrahim-mubarak — CHANGES_REQUESTED** [review]

File: lib/src/config_writer/vscode_config_writer.dart (hunk 1 above)
Comment (chillbrodev) [reply lib/src/config_writer/vscode_config_writer.dart:36]: Yes, in my env config I have urls and this regex was corrupting the urls by removing the slashes in the url. Hence why I added the Try/Catch block.

File: lib/src/config_writer/vscode_config_writer.dart (hunk 2 above)
Comment (chillbrodev) [reply lib/src/config_writer/vscode_config_writer.dart:71]: Sure thing, Thought I caught them all. My bad

File: lib/src/config_writer/vscode_config_writer.dart (hunk 1 above)
Comment (ibrahim-mubarak) [reply lib/src/config_writer/vscode_config_writer.dart:36]: My bad about this
I think we can update the regex to catch comments only.

Comment (ibrahim-mubarak) [reply lib/src/config_writer/vscode_config_writer.dart:36]: Can you try out this package?
https://pub.dev/packages/json5

Comment (chillbrodev) [reply lib/src/config_writer/vscode_config_writer.dart:36]: Sure, I'll give it a shot

**Pull request closed without merging**